types/src/transaction/mod.rs

```diff
@@ -1126,6 +1153,59 @@ impl VMValidatorResult {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub struct VMErrorDetail {
+    pub status_code: StatusCode,
+    pub message: Option<String>,
+}
+
+impl VMErrorDetail {
+    pub fn new(status_code: StatusCode, message: Option<String>) -> Self {
+        Self {
+            status_code,
+            message,
+        }
+    }
+
+    pub fn status_code(&self) -> StatusCode {
+        self.status_code
+    }
+
+    pub fn message(&self) -> &Option<String> {
+        &self.message
+    }
+}
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub struct TransactionAuxiliaryData {
+    pub detail_error_message: Option<VMErrorDetail>,
```

Comment:
Move part makes sense to me. My only question is how easy would it be in the future for us to add fields into this struct in the future. IIUC the struct will be stored in db using the derived ser/de trait? If so how can we add items here in the future?

Reply:
Adding new optional fields should be fine. the data can be deserialized to the new struct with **None** value for new fields.

Reply:
No, `bcs` doesn't do that. you need a `enum`, at least at least in the DB schema

Reply:
updated to enum